### ISSUE TITLE
Feature Request: add `match` option to `bs.reload` (like `bs.stream`)

### ISSUE THREAD

**Issue by mgcrea** [BrowserSync/browser-sync#894]
Would be great to add a `match` option to `bs.reload` like what can be done with `bs.stream`. The use case is to be able to easily filter out styles sourcemaps that triggers a reload (gulp workflow).

According to the source, it looks like it just might work (as reload defers to stream):

    .pipe(bs.reload({stream: true, match: '**/*.css'}));

But it does not, neither does:

    .pipe(bs.stream({match: '**/*.css'}));

Only the following works and triggers a proper livereload

    .pipe(gulpif('*.css', bs.reload({stream: true})));

Using `"browser-sync": "^2.10.0"`

**Comment by shakyShane** [comment]
Could you create a working example for me to test, like https://github.com/Browsersync/recipes/tree/master/recipes/gulp.ruby.sass

I've tested this use-case in many formats, so I'll be interested to see how it fails for you. Once I have that I can re-open the issue :)

**Comment by medifle** [comment]
## Version
```bash
gulp@3.9.0
browser-sync@2.10.0
gulp-if@2.0.0
gulp-sourcemaps@1.6.0
gulp-sass@2.1.1
```
Here is a isolated case for testing.
https://github.com/medifle/recipes/tree/master/browserSync

The `match` issue does exist when browserSync serves the dir whose name starts as `.`, i.e. `.tmp`.

`gulp-if`, `gulp-filter` handles this problem well, so if you try them the issue would not occur.

**Comment by shakyShane** [comment]
Thanks @medifle I look into this :)

**Comment by shakyShane** [comment]
fixed in browser-sync@2.10.1
